src/librustc_data_structures/profiling.rs

```diff
@@ -97,12 +97,27 @@ use std::time::{Duration, Instant};
 use measureme::{EventId, EventIdBuilder, SerializableString, StringId};
 use parking_lot::RwLock;
 
-/// MmapSerializatioSink is faster on macOS and Linux
-/// but FileSerializationSink is faster on Windows
-#[cfg(not(windows))]
-type SerializationSink = measureme::MmapSerializationSink;
-#[cfg(windows)]
-type SerializationSink = measureme::FileSerializationSink;
+cfg_if! {
+    if #[cfg(target_arch = "wasm32")] {
+        cfg_if! {
+            if #[cfg(target_os = "wasi")] {
+                type SerializationSink = measureme::FileSerializationSink;
+            } else {
+                type SerializationSink = measureme::ByteVecSink;
```

Comment:
Same here, `ByteVecSink` is the fallback. `FileSerializationSink` is used if `target_os` is `wasi` or `windows`. `MmapSerializationSink` get used on UNIX-like OSes.

Reply:
I accepted the code suggestion. Code suggestions are probably easiest to resolve these remaining points. I'm fighting a fever and it is hard for me to think.

Reply:
See below.